### ISSUE TITLE
Adding a FLUIDOS Node Helm installation which injects also the dependencies

### ISSUE THREAD

**Issue by fracappa** [fluidos-project/node#87]
Hello everyone.

@andreacv98 and I are planning to provide a FLUIDOS Node installation through Helm charts.

The idea is to use this helm package to allow user the FLUIDOS Node installation on K8s cluster already up and running.

We already have a [bash script](https://github.com/fluidos-project/node/blob/main/tools/scripts/setup.sh) which does the same task but we would like to provide a smoother and cross-environment approach.

Any feedback is highly appreciated.

Thanks,
Francesco

**Comment by frisso** [comment]
Definitely in favor of using Helm.
My preference goes as having Helm as main component, and a bash script to automate everything if you want to have an even simpler life. Bash, or Ansible, choose the one you prefer (or even both).

**Comment by fracappa** [comment]
I have been working on a single Helm installation for our project, but I've encountered some limitations that need to be addressed.

Helm allows for the installation of dependencies as sub-charts, ensuring that when a user runs `helm install ...`, the dependencies are also installed. However, this process installs all resources within the same namespace.

In our case, when running `helm install fluidos-node --namespace fluidos`, the Liqo resources are also installed in the fluidos namespace. This approach is not ideal because the liqoctl commands return errors due to the absence of the Liqo namespace.

Apparently, that the default Helm functionality does not support installing the main chart and its dependencies into different namespaces.

There is a third-party tool called Helmfile that addresses this issue, but I am uncertain whether introducing another dependency only for the installation is reasonable.

An alternative is to use a post-installation hook to install `liqo` in the correct namespace. This would involve a Kubernetes job that runs after the main chart installation. However, I have not tested this method, and I am unsure if it is a clean solution.

In summary, we have three potential approaches:

1.  Use Helmfile to install the fluidos-node and its dependencies into the appropriate namespaces.
2. Implement pre-install/post-install hooks to install the dependencies in different namespaces than the main chart.
3. Separate the installation of the main chart and its dependencies, requiring the user to install them individually.

I would appreciate any feedback you may have.

Regards, 
Francesco

**Comment by LorenzoMoro** [comment]
Does this multiple namespace issue occurs with Liqo only or does it involves other namespaces (currently and potentially in the future)? In the first case I would probably go with the third option, since Liqo could be installed as a standalone component (and could also be already installed in some scenarios), while in the last case I would probably go with helmfile, since I find it cleaner.

**Comment by fracappa** [comment]
Hi @LorenzoMoro , thanks for your feedback.

This issue does not occur only in the liqo case, but with others' as well.

For example, we also need to have `metrics-server` installed which is usually deployed within `kube-system` namespace.

Another example comes form the certificate manager (a component we need in the new version as we're going to use webhooks) , usually deployed within `cert-manager` namespace.

So, if we want to leave an easier installer to the user, which is also able to inject all the dependencies, [helmfile](https://github.com/helmfile/helmfile)  is definitely the best option.

If we want to provide an helm installer that is only in charge of the FLUIDOS Component (Node, meta-orchestrator, etc),  we can just use vanilla helm.

However, this last option requires the user to have all the other dependencies already installed.

Just bare in mind that when we already have a cluster with Liqo installed, this last one needs to be configured so it can use the FLUIDOS Node grpc server.

**Comment by LorenzoMoro** [comment]
Alright. The issue probably shifts towards the definition of what a FLUIDOS Component is and weather it makes sense outside the FLUIDOS Node or not. This is crystal clear for Liqo, but I'm not sure about the other components you listed, so I'm wondering if it would be better to have a namespace reassignment. Nevertheless, for those outside fluidos namespace for whatever reason, helmfile seems preferrable (I've seen it's quite maintained as well).

**Comment by fracappa** [comment]
I'm not sure I got your idea. What do you mean by "namespace reassignment"?

I think that asking the user to execute just one command to have a full FLUIDOS Node installation working could be a good achievement. 

In this prospective, we can both rely on helmfile or pre/post installation hooks: the first is a cleaner approach.

The talk about FLUIDOS Components union was to address what we have talked with @stefano81 .

If we want the orchestrator to have the FLUIDOS Node as a dependency, it should use the same helmfile approach otherwise it could miss some Node's dependencies (liqo, metrics-server, cert-manager).

On the other side, if we want to provide only FLUIDOS-related software installations (e.g. Node+orchestrator) we can just use helm as we wouldn't have the problem of running deployment into different namespaces (everything under `fluidos` namespace): this approach requires the user to install all the other dependencies.
